### PR TITLE
Guides pages for OBP

### LINKED ISSUE #2107: [ NEW GUIDE ] OBP intro
This guide should cover:

- overview of OBP package
- pip install instructions (and any specific package installation constraints)

Existing docs for reference:

- obp readme
- obp installation page

### LINKED ISSUE #2108: [ NEW GUIDE ] OBP get started
This guide should include:

- minimum working example of OBP addon

docs for reference:
- OBP tutorial (minus patterns)

### PR DESCRIPTION
**Pull request by kaelynj**
Closes #2107 and closes #2108 

I'll include the files in the toc and satiate CI once #2148 is merged to avoid dealing with conflicts.

### CODE REVIEW

**Comment by caleb-johnson** [comment]
Thanks Kaelyn, looks great! I'll just drop a few thoughts below for the getting started guide:

- Paragraph 1
    - For clarity can we change "slices" --> "circuit slices" when we first mention it since this is kind of a made-up term
    - I like that you highlight preparation of the slices and operator budget. I think we need to include the truncation budget here, as it is arguably more important than the operator budget (can explain this in more detail if you're interested). Some of the language in your OBP landing page intro could be moved over here regarding this.
    
- Prepare inputs to backpropagate
    - qiskit_addons_utils.slice_by_gate_types() --> qiskit_addon_utils.slicing.slice_by_gate_types

File: docs/guides/qiskit-addons-obp.mdx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+---
+title: Operator backpropagation (OBP)
+description: Learn about the operator backpropagation addon to reduce quantum resource overhead
```
Comment (caleb-johnson) [inline docs/guides/qiskit-addons-obp.mdx:3]:
```suggestion
description: Learn about the operator backpropagation addon to reduce the depth of quantum circuits
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+---
+
+# Operator backpropagation (OBP)
+
+Operator backpropagation (OBP) is a technique to reduce circuit depth by trimming operations from its end at the cost of more operator measurements. There are a number of ways in which operator backpropagation can be performed, and this package uses a method based on Clifford perturbation theory.
+
+As one propagates an operator further through a circuit, the size of the additional observable needed for measurement grows exponentially. This results in both a classical and quantum resource overhead. However, for some circuits, the resulting distribution of additional Pauli observables is more concentrated than the worst-case exponential scaling. This implies that some terms in an observable which small coefficients can be truncated to reduce the quantum overhead. The error incurred by doing so can be controlled to find a suitable tradeoff between precision and efficiency.
```
Comment (caleb-johnson) [inline docs/guides/qiskit-addons-obp.mdx:10]:
```suggestion
As one propagates an operator further through a circuit, the size of the observable to measure grows exponentially. This results in both a classical and quantum resource overhead. However, for some circuits, the resulting distribution of additional Pauli observables is more concentrated than the worst-case exponential scaling. This implies that some terms in an observable with small coefficients can be truncated to reduce the quantum overhead. The error incurred by doing so can be controlled to find a suitable tradeoff between precision and efficiency.
```

**Review by caleb-johnson — COMMENTED** [review]
Looks great, just a couple of minor comments/requests

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 3]
```diff
@@ -0,0 +1,322 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "0735e07c-1289-4326-8b1b-11ad53b00a32",
+   "metadata": {},
+   "source": [
+    "# Get started with OBP\n",
+    "\n",
+    "Preparing a quantum workload with operator backpropagation requires that first, a selection of \"slices\" be made and second, an upper bound to the size of the backpropagated operator should be set. During backpropagation, the number of terms in the operator of an $N$ qubit circuit will approach $4^N$ quickly in the worst case scenario. In this page we will overview the steps involved in applying OBP to a quantum workload.\n",
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:10]:
```suggestion
    "When you prepare a quantum workload with operator backpropagation (OBP), first you must make a selection of \"slices\", and second, you must set an upper bound to the size of the backpropagated operator. During backpropagation, the number of terms in the operator of an $N$-qubit circuit will approach $4^N$ quickly in the worst-case scenario. This guide demonstrates the steps involved in applying OBP to a quantum workload.\n",
```

**Comment by BryceFuller** [comment]
### I have a few suggested edits to `qiskit-addons-obp.mdx`:

> This method has the benefit that the overhead incurred by backpropagating various gates is determined by the non-Cliffordness of that gate (i.e. how much of a particular portion of $U_C$ which contains non-Clifford instructions).

I would change to 

> This method has the benefit that the overhead incurred by backpropagating various gates scales with the non-Cliffordness of $U_C$ (i.e. how much of $U_C$ is comprised of non-Clifford instructions).


Also, later in that section I would append an additional sentence

> Each of these slices are then analytically applied in sequence to measure the back propagated operator $O'$ and may or may not contribute to the overall size of the sum, depending on if the slice is a Clifford vs non-Clifford operation.

would become 

> Each of these slices are then analytically applied in sequence to measure the back propagated operator $O'$ and may or may not contribute to the overall size of the sum, depending on if the slice is a Clifford vs non-Clifford operation. If an error budget is allocated, truncation will then occur between the application of each slice. 


### Here are some suggested edits to `qiskit-addons-obp-get-started.ipynb`:

there is a space missing in one of the print messages for the cell that runs the actual OBP method

> New observable has 17 terms, which can be combined into7 groups.

should print out

> New observable has 17 terms, which can be combined into 7 groups.

Also, the value of `t` here has been changed from 0.2 in the obp tutorial, to 2. This results in the optimal expval being very close to 0, and the final answers in the notebook are all within statistical accuracy from zero. Can you return `t` to it's earlier value so we know there is a larger expval.

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+    "\n",
+    "The main component of the `qiskit-addons-obp` package is the `backpropagate()` function. It ingests arguments for the final observable to reconstruct, a set of circuit slices to compute classically, and, optionally, a `TruncationErrorBudget` or `OperatorBudget` to provide constraints on the truncation that is done. Once these are specified, the classically computed backpropagated operator, $O'$ is calculated iteratively by applying the gates from each slice, $s$, in the following way:\n",
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:12]:
```suggestion
    "The main component of the `qiskit-addons-obp` package is the `backpropagate()` function. It ingests arguments for the final observable to reconstruct, a set of circuit slices to compute classically, and, optionally, a `TruncationErrorBudget` or `OperatorBudget` to provide constraints on the truncation that is done. Once these are specified, the classically computed backpropagated operator $O'$ is calculated iteratively by applying the gates from each slice, $s$, in the following way:\n",
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+    "\n",
+    "$$ O'^{(s)} = \\mathcal{U}_{S-s+1}^\\dagger O'^{(s-1)} \\mathcal{U}_{S-s+1} $$\n",
+    "\n",
+    "where $S$ is the total number of slices and $\\mathcal{U}_{s}$ represents a single slice of the circuit. For this example, we will also be using the `qiskit-addons-utils` package to prepare the circuit slices as well as generate the example circuit we'll use.\n",
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:16]:
```suggestion
    "where $S$ is the total number of slices and $\\mathcal{U}_{s}$ represents a single slice of the circuit. This example uses the `qiskit-addons-utils` package to prepare the circuit slices as well as generate the example circuit.\n",
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+    "\n",
+    "\n",
+    "To begin, the problem we'll generate is the time evolution of a Heisenberg XYZ chain. This Hamiltonian has the form\n",
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:19]:
```suggestion
    "To begin, consider the time evolution of a Heisenberg XYZ chain. This Hamiltonian has the form\n",
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+    "\n",
+    "$$ \\hat{H} = \\sum_{(j,k)} \\left( J_xX_jX_k + J_yY_jY_k + J_z Z_jZ_k \\right) + \\sum_{j} \\left(h_xX_j + h_yY_j + h_zZ_j\\right)$$\n",
+    "\n",
+    "and the expectation value to measure will be $\\langle Z_0 \\rangle$.\n",
+    "\n",
+    "The following code snippet will generate the Hamiltonian in the form of a `SparsePauliOp` by using the `qiskit_addons_utils.problem_generators` module and a `CouplingMap`. We'll set the coupling constants to $J_x=\\pi/8$, $J_y=\\pi/4$, $J_z=\\pi/2$ and external magnetic fields to $h_x=\\pi/3$, $h_y=\\pi/6$, $h_z=\\pi/9$ and then generate a circuit which models its time evolution."
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:25]:
```suggestion
    "The following code snippet generates the Hamiltonian in the form of a `SparsePauliOp` by using the `qiskit_addons_utils.problem_generators` module and a `CouplingMap`. Set the coupling constants to $J_x=\\pi/8$, $J_y=\\pi/4$, $J_z=\\pi/2$ and external magnetic fields to $h_x=\\pi/3$, $h_y=\\pi/6$, $h_z=\\pi/9$, and then generate a circuit that models its time evolution."
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6250fbf9-bfe8-41cf-b2e3-bec7ba7020f7",
+   "metadata": {},
+   "source": [
+    "### Prepare inputs to backpropagate\n",
+    "\n",
+    "Next we'll generate the circuit slices we'll use to backpropagate. In general, the choice of how to slice can have an impact on how well backpropagation performs for a given problem. Here we will group gates of the same type into slices using the `qiskit_addons_utils.slice_by_gate_types()` function."
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:94]:
```suggestion
    "Next, generate the circuit slices for backpropagation. In general, the choice of how to slice can have an impact on how well backpropagation performs for a given problem. Here, group gates of the same type into slices using the `qiskit_addons_utils.slice_by_gate_types()` function."
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+    "slices = slice_by_gate_types(circuit)\n",
+    "print(f\"Separated the circuit into {len(slices)} slices.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a9ef8832-352c-4f2f-a8f5-96a2c8207fd1",
+   "metadata": {},
+   "source": [
+    "Once the slices have been generated, we will then specify an `OperatorBudget` in order to provide the `backpropagate()` function a condition to stop back propagating the operator and prevent the classical overhead from growing further. We can also specify a truncation error budget for each slice wherein Pauli terms with small coefficients will be truncated from each slice until the error budget is filled. Any leftover budget will then be added to the following slice's budget.\n",
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:121]:
```suggestion
    "Once the slices have been generated, specify an `OperatorBudget` to provide the `backpropagate()` function with a condition to stop backpropagating the operator and prevent the classical overhead from growing further. You can also specify a truncation error budget for each slice wherein Pauli terms with small coefficients will be truncated from each slice until the error budget is filled. Any leftover budget will then be added to the following slice's budget.\n",
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+    "\n",
+    "Here we specify that backpropagation should stop when the number of qubit-wise commuting Pauli groups in the operator grows past $8$ and allocate an error budget of $0.005$ for each slice."
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:123]:
```suggestion
    "Here, specify that backpropagation should stop when the number of qubit-wise commuting Pauli groups in the operator grows past $8$, and allocate an error budget of $0.005$ for each slice."
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 11]
```diff
@@ -0,0 +1,322 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d649dd72-0c53-4ea6-9c14-4828db2526a6",
+   "metadata": {},
+   "source": [
+    "### Backpropagate slices\n",
+    "\n",
+    "Here we'll define the final observable to measure and run the backpropagation across each slice. The `backpropagate()` function will return three outputs: the backpropagated observable, the remaining circuit slices which were note backpropagated (and which should be run on quantum hardware), and metadata about the backpropagation.\n",
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:144]:
```suggestion
    "In this step you will define the final observable to measure and run the backpropagation across each slice. The `backpropagate()` function returns three outputs: the backpropagated observable, the remaining circuit slices that were not backpropagated (and which should be run on quantum hardware), and metadata about the backpropagation.\n",
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 12]
```diff
@@ -0,0 +1,322 @@
...
+    "\n",
+    "It should be noted here that both the `OperatorBudget` and the `TruncationErrorBudget` are optional parameters for the `backpropagate()` method and, in general, the best choice for both should be heuristically chosen and requires some amount of experimentation.\n",
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:146]:
```suggestion
    "Note that both the `OperatorBudget` and the `TruncationErrorBudget` are optional parameters for the `backpropagate()` method. In general, the best choice for both should be heuristically chosen and requires some amount of experimentation.\n",
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 13]
```diff
@@ -0,0 +1,322 @@
...
+   "execution_count": 4,
+   "id": "100bdf25-25c4-466f-9cdc-c85ee93e646f",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Backpropagated 7 slices.\n",
+      "New observable has 17 terms, which can be combined into7 groups.\n",
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:164]:
```suggestion
      "New observable has 17 terms, which can be combined into 7 groups.\n",
```

File: docs/guides/qiskit-addons-obp-get-started.ipynb [hunk 14]
```diff
@@ -0,0 +1,322 @@
...
...iL9VE4yU04ys49ec5N9AuVkf8D5ySCCQg+9P5sR377NoI2vkvjmeixl5QAkLtlA3/dnc/FERvVhOTVNU0NCQoJq87JVTbkVHE/DarHQbf5ETny0DWt5RY3T1KLH3IR9HNVLs68Xbr6eDvlsW7n5emL29XLIZxs1N2fMTGhDjX6qtTw48nt/LclMOTUy83EDb5VOWvZ2q5yfo+ltfSv7BMrJ/oBxyDURFKo6l6f5oz14YMkUsvf8xJW8ixQkp+PTtCGpP9/bmZ8P1blxWn1VW26JizfQ76M57PnjyurX1jRNCD3wCPRj9L7llBWWaFaD2dcLj0A/zeZvD61zc8bMhHGptTwY6XsvmSnXwAyb+0FRuePn5eNWOb/6RvYJlJP9AeOQQQQ7pX6+h/BhPekwfQT75n5AxPiBJK9NIDL6Uf475S2AGqepYdSoUarNS6kbcyvKyKM4M++619Q0TQ16zk0o48heegT6GeZH5o0kN+VkvWEsavXTSMuDZKacWpk1MBtr516v61vZJ1BO9gecn5zOcAcOLlhH68cewjs0mLv7dyZx0QauFpUQ1D4cNx/Pm6appXv37qrNyx5VuXk1CtC6lOvoPTdHWrVqFTExMcTExBAaGsrChQuve3z+/PnrXj9nzhw+/fRTJk2axOjRozl27FiNn7t48WKysrL49ttviYmJYejQoRw8eJAtW7awfft2h/099bmXd0JyU04yMxbpp3KSmXKSmX30nJvsEygn+wPOTQYRFIjv+nsunsysfnw5JZuP24zn3nEPk/TeFrBaOfxWPJ1mjKFdzLCbpqll1qxZqs3LFrXlVpJboGldN9JbbmqKjo5mxYoVREVF8eabb/LCCy9c9zg4OLj6tXl5eZjNZoYPH87q1av505/+xFdffcWmTZto164dw4cP5/7776e8vJysrCxCQkLo1asXK1asIDY2lkOHDjFo0CA++eQTh/099bmXd0JyU04yMxbpp3KSmXKSmX30lJvsEygn+wPGIqcz1IGq+zkDFJ3NY/vTC697vqZp4hdbH59v0zThWN9//z2nTp0iLi6uxsdVEhMTadOmDQAWi4Xly5czd+5csrKyePbZZ2ndujW5ubmUlpbi6+tb/b5Vq1axevVqPvzwQ1xcXDh37pzKf6EQQgghhOPIPoH9ZH/AuciRCAbUsmVLrUtwSvU5t5ycHN5++20WL
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb]:
```suggestion
    "### Transpile and execute the quantum workload\n",
    "\n",
    "Now that you have backpropagated the operator, you can execute the remaining portion of the circuit on a QPU. The quantum workload, using the Estimator, should include the `bp_circuit_trunc` circuit and must measure the backpropagated operator `backpropagated_observable`.\n",
    "\n",
    "The following code snippet transpiles both the original and backpropagated circuit, and runs them on a QPU so you can compare the results."
```

Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:300]:
```suggestion
  "description": "Get started using the operator backpropagation Qiskit addon (OBP)",
```

Comment (abbycross) [inline docs/guides/qiskit-addons-obp-get-started.ipynb:318]:
```suggestion
  "title": "Get started with OBP"
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+## Installation
+
+There are two ways to install the OBP package. Via PyPI and building from source. We also recommend installing these packages in a [virtual environment](https://docs.python.org/3.10/tutorial/venv.html) to ensure separation between package dependencies.
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:14]:
```suggestion
You can install the OBP package in one of two ways: via PyPI or building from source. Consider installing these packages in a [virtual environment](https://docs.python.org/3.10/tutorial/venv.html) to ensure separation between package dependencies.
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+The most straightforward way to install the `qiskit-addon-obp` package is via PyPI.
+
+```bash
+pip install qiskit-addon-obp
+```
+
+### Build from source
+
+Users who wish to contribute to this package or who want to install it manually may do so by first cloning the repository
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:26]:
```suggestion
Users who wish to contribute to this package or who want to install it manually may do so by first cloning the repository:
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+
+```bash
+git clone git@github.com:Qiskit/qiskit-addon-obp.git
+```_
+
+and install the package via `pip`. The repository also contains example notebooks which can be run. And if you plan on developing in the repository, you may want to also install the `dev` dependencies.
+
+Adjust the options to suit your needs
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:34]:
```suggestion
and install the package via `pip`. The repository also contains example notebooks. If you plan on developing in the repository, install the `dev` dependencies.

Adjust the options to suit your needs:
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+
+```bash
+pip install tox notebook -e '.[notebook-dependencies, dev]'
+```
+
+## Theoretical background
+
+When using the Estimator primitive, the output of a quantum workload is the estimation of one or more expectation values $\langle O \rangle$ with respect to some state prepared using a QPU. In this section we will summarize the procedure.
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:42]:
```suggestion
When using the Estimator primitive, the output of a quantum workload is the estimation of one or more expectation values $\langle O \rangle$ with respect to some state prepared using a QPU. This section summarizes the procedure.
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+
+We first start by writing the expectation value measurement of an observable $O$ in terms of some initial state $|\psi\rangle$ and a quantum circuit $U_Q$:
+
+$$ \langle O \rangle_{U|\psi\rangle} = \langle\psi | U^\dagger O U |\psi \rangle. $$
+
+To distribute this problem across both classical and resources, we can split the circuit $U$ into two subcircuits, $U_C$ and $U_Q$, classically simulate the circuit $U_C$, then execute the circuit $U_Q$ on quantum hardware and use the results of the classical simulation to reconstruct the measurement of the obervable $O$.
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:48]:
```suggestion
First, start by writing the expectation value measurement of an observable $O$ in terms of some initial state $|\psi\rangle$ and a quantum circuit $U_Q$:

$$ \langle O \rangle_{U|\psi\rangle} = \langle\psi | U^\dagger O U |\psi \rangle. $$

To distribute this problem across both classical and quantum resources, split the circuit $U$ into two subcircuits, $U_C$ and $U_Q$, classically simulate the circuit $U_C$, then execute the circuit $U_Q$ on quantum hardware and use the results of the classical simulation to reconstruct the measurement of the observable $O$.
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+
+![OBP diagram depicting splitting a circuit into two subcircuits, classically computing one of the subcircuits, then measuring the other circuit using quantum hardware](/images/guides/qiskit-addons/obp-diagram.png)
+
+
+The subcircuit $U_C$ should be selected to be classically simulable and will compute the expectation value
+
+$$ \langle O' \rangle \equiv U_C^\dagger O U_C $$
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:55]:
```suggestion
$$ \langle O' \rangle \equiv U_C^\dagger O U_C, $$
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+
+which is the version of the initial operator, $O$ evolved through the circuit $U_C$. Once $O'$ has been determined, the quantum workload is prepared wherein the state $|\psi\rangle$ is initiated, has the circuit $U_Q$ applied to it, and then measures the expectation value $O'$. We can show that this is equivalent to measuring $\langle O \rangle$ by writing:
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:57]:
```suggestion
which is the version of the initial operator $O$ evolved through the circuit $U_C$. Once $O'$ has been determined, the quantum workload is prepared wherein the state $|\psi\rangle$ is initiated, has the circuit $U_Q$ applied to it, and then measures the expectation value $O'$. You can show that this is equivalent to measuring $\langle O \rangle$ by writing:
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+
+$$ \langle \psi | U_Q^\dagger O' U_Q \psi \rangle = \langle \psi | U_Q^\dagger U_C^\dagger O U_CU_Q \psi \rangle = \langle\psi | U^\dagger O U |\psi \rangle = \langle O \rangle_{U|\psi\rangle}$$
+
+
+Lastly, in order to measure the expectation value $\langle O' \rangle$, we must require it to be decomposable into a sum of Pauli strings
+
+$$ O' = \sum_P c_P P $$
+
+where $c_P$ are real coefficients of the decomposition and $P$ is some Pauli string composed of $I$, $X$, $Y$, and $Z$ operators. This ensures that we can reconstruct the expectation value of $O$ by
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:66]:
```suggestion
Lastly, in order to measure the expectation value $\langle O' \rangle$, we must require it to be decomposable into a sum of Pauli strings

$$ O' = \sum_P c_P P, $$

where $c_P$ are real coefficients of the decomposition and $P$ is some Pauli string composed of $I$, $X$, $Y$, and $Z$ operators. This ensures that you can reconstruct the expectation value of $O$ by
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+
+$$ \langle \psi | U_Q^\dagger O' |\psi \rangle = \sum_P c_P \langle \psi | U_Q^\dagger P U_Q | \psi \rangle. $$
+
+
+### Truncating terms
+
+This scheme offers a tradeoff between the required circuit depth of $U_Q$, the number of circuit executions on quantum hardware, and the amount of classical computing resources needed to compute $O'$. In general, as you choose to backpropagate further through a circuit, the number of Pauli strings to measure as well as the error-mitigation overhead both grow exponentially (alongside the classical resources needed to simulate $U_C$).
+
+Fortunately, the decomposition of $O'$ can often contain coefficients which are quite small and can be truncated from the final measurements used to reconstruct $O$ without incurring much error. The `qiskit-addon-obp` package possesses functionality to specify an error budget which can be used to automatically search for terms which can be truncated, to within some error tolerance.
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:75]:
```suggestion
Fortunately, the decomposition of $O'$ can often contain coefficients that are quite small and can be truncated from the final measurements used to reconstruct $O$ without incurring much error. The `qiskit-addon-obp` package possesses functionality to specify an error budget, which can automatically search for terms that can be truncated, to within some error tolerance.
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+
+### Clifford perturbation theory
+
+Lastly, the `qiskit-addon-obp` package approaches operator backpropagation based on Clifford perturbation theory. This method has the benefit that the overhead incurred by backpropagating various gates is determined by the non-Cliffordness of that gate (i.e. how much of a particular portion of $U_C$ which contains non-Clifford instructions).
+
+This approach to OBP begins by splitting the simulated circuit, $U_C$, into *slices*
+
+$$ U_C = \prod_{s=1}^S \mathcal{U}_s = \mathcal{U}_S...\mathcal{U}_2\mathcal{U}_1 $$
+
+where $S$ represents the total number of slices and $\mathcal{U}_s$ denotes a single slice of the circuit $U_C$. Each of these slices are then analytically applied in sequence to measure the back propagated operator $O'$ and may or may not contribute to the overall size of the sum, depending on if the slice is a Clifford vs non-Clifford operation.
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:86]:
```suggestion
Lastly, the `qiskit-addon-obp` package approaches operator backpropagation based on Clifford perturbation theory. This method has the benefit that the overhead incurred by backpropagating various gates is determined by the non-Cliffordness of that gate (as in, how much of a particular portion of $U_C$ contains non-Clifford instructions).

This approach to OBP begins by splitting the simulated circuit, $U_C$, into *slices*:

$$ U_C = \prod_{s=1}^S \mathcal{U}_s = \mathcal{U}_S...\mathcal{U}_2\mathcal{U}_1, $$

where $S$ represents the total number of slices and $\mathcal{U}_s$ denotes a single slice of the circuit $U_C$. Each of these slices are then analytically applied in sequence to measure the backpropagated operator $O'$, and might or might not contribute to the overall size of the sum, depending on if the slice is a Clifford versus non-Clifford operation.
```

File: docs/guides/qiskit-addons-obp.mdx [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+
+
+
+## Next steps
+
+<Admonition type="tip" title="Recommendations">
+    - Get started with OBP
+    - Become familiar with the error mitigation techniques available in Qiskit Runtime
```
Comment (abbycross) [inline docs/guides/qiskit-addons-obp.mdx:94]:
```suggestion
    - [Get started with OBP.](/guides/qiskit-addons-obp-get-started.ipynb)    
    - Become familiar with the [error mitigation techniques](/guides/error-mitigation-and-suppression-techniques) available in Qiskit Runtime.
```

**Review by abbycross — APPROVED** [review]
copyediting suggestions, otherwise good to go

**Comment by jenglick** [comment]
@kaelynj -- pending getting [Bryce's suggestions](https://github.com/Qiskit/documentation/pull/2248#issuecomment-2468448660) folded in, are you also able to also add in the results for running the backprop and backprop w truncation circuits run on the StatevectorEstimator to isolate the OBP-specific effects? Beyond that, this looks good to me!

**Comment by kaelynj** [comment]
> @kaelynj -- pending getting [Bryce's suggestions](https://github.com/Qiskit/documentation/pull/2248#issuecomment-2468448660) folded in, are you also able to also add in the results for running the backprop and backprop w truncation circuits run on the StatevectorEstimator to isolate the OBP-specific effects? Beyond that, this looks good to me!

This should be included now (along with @BryceFuller's  suggestion).

@Eric-Arellano I'm having an issue with CI on the `checkPatternsIndex` script. NPM claims that the pages are not present in the toc when they actually are. Could this be because the pages are nested one level further than the rest of ours?

**Comment by jenglick** [comment]
@kaelynj The tutorial for OBP in Learning Platform just dropped -- would you be able to add a link to this in the "next steps" section of the getting started page?

https://learning.quantum.ibm.com/tutorial/improving-estimation-of-expectation-values-with-operator-backpropagation

**Comment by kaelynj** [comment]
> @kaelynj The tutorial for OBP in Learning Platform just dropped -- would you be able to add a link to this in the "next steps" section of the getting started page?
> 
> https://learning.quantum.ibm.com/tutorial/improving-estimation-of-expectation-values-with-operator-backpropagation

Ended up adding this link in the AQC pr #2291 to save time for QDC

**Pull request merged**